Remove warnings about autocreated index fields

scollazo opened the pull request:
Removes models.W042 warning when running manage.py commands

Connected to https://github.com/archivematica/Issues/issues/1759

scollazo committed:
Set DEFAULT_AUTO_FIELD

Removes models.W042 warning when running manage.py commands

replaceafill commented on src/archivematica/dashboard/settings/base.py:
```diff
@@ -709,3 +709,6 @@ def _get_settings_from_file(path):
 globals().update(email_settings.get_settings(config))
+
+# Remove models.W042 warning
+DEFAULT_AUTO_FIELD = "django.db.models.AutoField"
```
I'd prefer to have a way to customize the [SILENCED_SYSTEM_CHECKS](https://docs.djangoproject.com/en/4.2/ref/settings/#silenced-system-checks) setting from configuration. That way we could turn the warning off in production but keep it for development in case we add new models.

scollazo committed:
Use SILENCED_SYSTEM_CHECKS in production settings

This removes the warnings about autocreated primary keys when
running some management commands.
